var/spack/repos/builtin/packages/vapor/package.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+import re
+
+from spack.package import *
+
+
+class Vapor(CMakePackage):
+    """VAPOR is the Visualization and Analysis Platform for Ocean,
+    Atmosphere, and Solar Researchers. VAPOR provides an interactive 3D
+    visualization environment that can also produce animations and
+    still frame images.
+    """
+
+    homepage = "https://www.vapor.ucar.edu"
+    url = "https://github.com/NCAR/VAPOR/archive/refs/tags/v3.9.0.tar.gz"
+    git = "https://github.com/NCAR/VAPOR.git"
+
+    maintainers("vanderwb")
+
+    version("main", branch="main")
+    version(
+        "3.9.0",
+        sha256="343ababe40b5824ef826f16c935a6dc1fb18e1a4c88ef967c8d64386f28a99a3",
+        preferred=True,
+    )
+
+    variant("doc", default=True, description="Build docs using Doxygen")
+    variant("ospray", default=False, description="Enable OSPRay raytracing")
+
+    depends_on("cmake@3.17:", type="build")
+    depends_on("xz")
+    depends_on("zlib")
+    depends_on("openssl")
+    depends_on("expat")
+    depends_on("curl")
+    depends_on("which")
+    depends_on("mesa-glu")
+    depends_on("gl")
+    depends_on("libxtst")
+    depends_on("libxcb")
+    depends_on("xcb-util")
+    depends_on("libxkbcommon")
+    depends_on("libpng")
+    depends_on("assimp")
+    depends_on("netcdf-c~dap~byterange")
+    depends_on("udunits")
+    depends_on("freetype")
+    depends_on("proj@:7")
+    depends_on("libgeotiff")
+    depends_on("python+ssl", type="build")
+    depends_on("py-numpy@1.21", type="build")
+    depends_on("py-scipy", type="build")
+    depends_on("py-matplotlib", type="build")
```

Comment:
Can you explain how these dependencies are "build" type only?

Reply:
These end up getting copied into the install prefix because Vapor has a requirement that they are integrated into that directory (see lines 111-121 of this recipe). They are technically needed at runtime, but the original installations are not used. I am hoping that the developers make this friendlier to external Python usage in the future, but that's how it works for now.

If it's better to keep this as build-link dependencies (or build-link-run), I'm happy to make that change.